portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -36,10 +41,26 @@
 
 @WebServlet("/comments")
 public class CommentsServlet extends HttpServlet {
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Query query = new Query("Comment");
+        List<Filter> filters = new ArrayList<>();
+        String[] filterOptions = request.getParameterMap().get("option");
+        Query query = new Query("Comment"); //No filters if nothing is selected;
+
+        //Add filters if users select one or more filtering options.
+        if(filterOptions != null){ 
+            if (filterOptions.length == 1){  
+                query = new Query("Comment").setFilter
+                    (new FilterPredicate("category", FilterOperator.EQUAL, filterOptions[0]));
+            }else {  
+                //if multiple filters are applied, use compositefilter
+                for (String option: filterOptions) {
+                    filters.add (new FilterPredicate("category", FilterOperator.EQUAL, option));
+                }
+                CompositeFilter optionsFilter = CompositeFilterOperator.or(filters);
```

Comment:
what happens if you create a composite.or of a single filter?

If the semantics are what you want, I would suggest removing the special case for == 1.

If not, add a comment for why we need a special case for == 1.

As a reader, I would expect CompositeFilterOperator.or() with a single element to correctly filter for that element.

Reply:
When I tried last week composite filter can only accept more than one filter, I just added the comment!